imap_processing/mag/l2/mag_l2.py

```diff
@@ -4,26 +4,58 @@
 import xarray as xr
 
 from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.mag import imap_mag_sdc_configuration_v001 as configuration
 from imap_processing.mag.constants import DataMode
+from imap_processing.mag.l1b.mag_l1b import calibrate_vector
 from imap_processing.mag.l2.mag_l2_data import MagL2
 
 
 def mag_l2(
-    calibration_dataset: xr.Dataset,
-    offset_dataset: xr.Dataset,
+    calibration_datasets: list[xr.Dataset],
+    offsets_dataset: xr.Dataset,
     input_data: xr.Dataset,
 ) -> list[xr.Dataset]:
     """
     Complete MAG L2 processing.
 
+    Processing uses 4 data input sources:
```

Comment:
Super helpful docstring! 

One small thing to fix: Below, you refer to the configuration yaml, where I think it should now be the python file.